File: examples/python/03_select_devices.py
import audio_sdk


def main() -> None:
    manager = audio_sdk.AudioManager()

    print("Discovered devices:")
    audio_sdk.print_devices(manager.enumerate_devices())

    input_device = audio_sdk.select_first_device(manager, audio_sdk.DeviceDirection.INPUT)
    output_device = audio_sdk.select_first_device(manager, audio_sdk.DeviceDirection.OUTPUT)
    print(f"Selected input device: {input_device.name} ({input_device.id})")
    print(f"Selected output device: {output_device.name} ({output_device.id})")


if __name__ == "__main__":
    main()

File: examples/cpp/03_select_devices.cpp
#include <iostream>

#include "example_utils.h"

int main() {
    audio_sdk::AudioManager manager;

    std::cout << "Discovered devices:\n";
    audio_sdk_examples::PrintDevices(manager.EnumerateDevices());

    const bool input_selected =
        audio_sdk_examples::SelectFirstDevice(manager, audio_sdk::DeviceDirection::kInput);
    const bool output_selected =
        audio_sdk_examples::SelectFirstDevice(manager, audio_sdk::DeviceDirection::kOutput);

    return input_selected && output_selected ? 0 : 1;
}
